sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -985,6 +985,23 @@ public void deleteDataset(String projectId, String datasetId)
           // impossible to insert into BigQuery, and so we send it out through the dead-letter
           // queue.
           if (nextRowSize >= maxRowBatchSize) {
+            Boolean isRetryAlways = false;
+            try {
+              // We verify whether the retryPolicy parameter is "retryAlways". If it is, then
+              // it will return true. Otherwise it will return false, or it may throw an NPE,
+              // which we need to catch and ignore.
+              isRetryAlways = retryPolicy.shouldRetry(null);
+            } catch (NullPointerException e) {
+              // We do not need to do anything about this exception, as it was expected;
```

Comment:
When would it throw an NPE ? If so I think you are probably not using the spec of that method correctly.